librubyfmt/src/render_targets.rs

```diff
@@ -3,11 +3,58 @@ use crate::line_tokens::LineToken;
 use crate::types::{ColNumber, LineNumber};
 use std::collections::HashSet;
 
+fn insert_at<T>(idx: usize, target: &mut Vec<T>, input: &mut Vec<T>) {
+    let drain = input.drain(..);
+    let mut idx = idx;
+    for item in drain {
+        target.insert(idx, item);
+        idx += 1;
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
 pub enum ConvertType {
     MultiLine,
     SingleLine,
 }
 
+pub trait LineTokenTarget {
+    fn push(&mut self, lt: LineToken);
+    fn insert_at(&mut self, idx: usize, tokens: &mut Vec<LineToken>);
+    fn into_tokens(self, ct: ConvertType) -> Vec<LineToken>;
+    fn last_token_is_a_newline(&self) -> bool;
+    fn index_of_prev_hard_newline(&self) -> Option<usize>;
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct BaseQueue {
+    tokens: Vec<LineToken>,
+}
+
+impl LineTokenTarget for BaseQueue {
+    fn push(&mut self, lt: LineToken) {
+        self.tokens.push(lt)
+    }
+
+    fn insert_at(&mut self, idx: usize, tokens: &mut Vec<LineToken>) {
```

Comment:
It looks like this is mostly called with a single element, should this just take that element instead of a vec?